pinot/net.py

```diff
@@ -56,28 +112,21 @@ def condition(self, g):
         # get the parameters
         theta = self.forward(g)
 
+        # put theta through noise model
+        theta = self.noise_model(theta)
+
         # parameterize the distribution according to the parameters
         # and the distribution class
-
-        distribution = self.distribution_class(
-            *self.param_transform(*torch.unbind(theta, dim=-1))
-        )
+        distribution = self.distribution_class(*theta)
 
         return distribution
 
-    def sample(self, g, n_samples=0):
-        """ Parameterize a distribution and sample from it.
-        """
-        distribution = self.condition(g)
-
-        if n_samples == 0:
-            return distribution.sample()
-
-        return distribution.sample_n(n_samples)
-
     def expectation(self, g):
```

Comment:
Why do we need this expectation thing?

Reply:
I thought it would be interesting to keep the functionality to enable training on mean estimate.

Reply:
well you can always do loss.mean() after you return the loss, no need to write an extra function in here for that, no?

Reply:
maybe I am missing the point here...